### PR TITLE
#11 - Add Github actions

### PR DESCRIPTION
**Pull request by rblalock**
- Publish NPM package
- Publish GH page

### CODE REVIEW

File: .github/workflows/publish-ghpages.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  publish-ghpages:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Predeploy GH Pages
+        run: npm run predeploy
```
Comment (keegandonley) [inline .github/workflows/publish-ghpages.yml:16]: Since predeploy does an `npm install` are you going to want to cache the node_modules here too?

Comment (rblalock) [reply .github/workflows/publish-ghpages.yml:16]: @keegandonley yeh good question.  So how does that caching work from different workflows in GH?  I noticed in our UIC one there are different workflows but one doesn't catch.  What determines the order of operations here?

Comment (keegandonley) [reply .github/workflows/publish-ghpages.yml:16]: @rblalock Yeah I see the one that's missing it. It's an old workflow and needs to be updated, they should all have it. The cache key is unique to the workflow so you'll need to add the caching step here too if you want to use it

Comment (rblalock) [reply .github/workflows/publish-ghpages.yml:34]: Ok makes sense.  @keegandonley Better? (just committed updates)

File: .github/workflows/publish-master.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+      - name: Set up Node.js for NPM
+        uses: actions/setup-node@v1
+        with:
+          registry-url: "https://registry.npmjs.org"
+          scope: '@pinpt'
+          node-version: '10.x'
+
+      - name: Set up Node.js for NPM
```
Comment (keegandonley) [inline .github/workflows/publish-master.yml:15]: @rblalock Are these two `Set up Node.js for NPM` steps different somehow or do we only need one?

File: .github/workflows/publish-master.yml [hunk 3]
```diff
@@ -12,14 +12,6 @@ jobs:
...
 
-
-      - name: Set up Node.js for NPM
-        uses: actions/setup-node@v1
-        with:
-          registry-url: "https://registry.npmjs.org"
-          scope: '@pinpt'
-          node-version: '10.x'
-
       - name: Set up Node.js for NPM
```
Comment (rblalock) [inline .github/workflows/publish-master.yml]: @keegandonley this was duplicated (it is in UIC too).  I assume it shouldn't be.

Comment (keegandonley) [reply .github/workflows/publish-master.yml]: Yeah don't think so

File: .github/workflows/publish-ghpages.yml (hunk 1 above)
Comment (keegandonley) [reply .github/workflows/publish-ghpages.yml:34]: Looks good 👍

File: .github/workflows/publish-master.yml (hunk 2 above)
Comment (rblalock) [reply .github/workflows/publish-master.yml:15]: Honestly i just copy pasta'd from our UIC one.  We don't need it IMO

**Review by keegandonley — APPROVED** [review]
🔥 🔥 🔥

File: .github/workflows/publish-master.yml (hunk 2 above)
Comment (keegandonley) [reply .github/workflows/publish-master.yml:15]: Oh sure blame it on me 🙄  🤣

**Pull request merged**